python/pyspark/sql/tests.py

```diff
@@ -429,6 +429,11 @@ def test_udf_with_input_file_name(self):
         row = self.spark.read.json(filePath).select(sourceFile(input_file_name())).first()
         self.assertTrue(row[0].find("people1.json") != -1)
 
+    def test_udf_shouldnt_accept_noncallable_object(self):
+        from pyspark.sql.functions import udf
+
+        self.assertRaises(TypeError, udf)
```

Comment:
I think this should have positive tests for a function and a class that defines `__call__`.